ngx_http_sxg_filter_module.c

```diff
@@ -346,13 +358,22 @@ static bool invoke_subrequests(ngx_str_t* link, ngx_http_request_t* req,
 }
 
 static ngx_int_t ngx_http_sxg_header_filter(ngx_http_request_t* req) {
-  // called every HTTP requests
+  // Called on every HTTP request.
   ngx_http_sxg_srv_conf_t* ssc =
       ngx_http_get_module_srv_conf(req, ngx_http_sxg_filter_module);
   if (!ssc->enable) {
     return ngx_http_next_header_filter(req);
   }
 
+  if (ssc->cert_path.len > 0 && req->uri.len == ssc->cert_path.len &&
+      ngx_memcmp(req->uri.data, ssc->cert_path.data, req->uri.len) == 0) {
+    int rc = ngx_http_send_header(req);
+    if (rc == NGX_ERROR || rc > NGX_OK || req->header_only) {
+        return rc;
+    }
+    return NGX_ERROR;
```

Comment:
This logic confuses me -- if the certificate matches and `ngx_http_send_header` returns `NGX_OK` this returns `NGX_ERROR`?

Reply:
In this case, the request is actually handled by `ngx_http_cert_chain_handler` so this returned value is ignored. But as you said, `NGX_OK` will match.